File: JavaScript/src/beggars.js
// https://www.codewars.com/kata/59590976838112bfea0000fa/javascript

export class Beggars
{
	
	/** @param {number[]} values
	 * @param {number} n */
	static beggars(values, n)
	{
		const arr = [];
		
		for (let i = 0; i < n; i++)
		{
			let res = 0;
			for (let j = i; j < values.length; j += n)
			{
				res += values[j];
			}
			arr.push(res);
		}
		
		return arr;
	}
}

File: Scala/src/main/scala/Beggars.scala
// https://www.codewars.com/kata/59590976838112bfea0000fa/scala

object Beggars
{
	def beggars(values: List[Int], n: Int): List[Int] =
	{
		var arr: List[Int] = List()
		for (i <- 0 until n)
		{
			var res = 0
			for (j <- i until values.size by n)
			{
				res += values(j)
			}
			arr :+= res
		}
		
		arr
	}
}
